src/Home.jsx

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect } from 'react'
+import { withRouter } from 'react-router'
+import ImageContainer from './components/ImageContainer'
+import { useGlobalContext } from './store/global'
+import queryString from 'query-string'
+import { searchPhotos, getAllPhotos } from './utils/data'
+
+const Home = (props) => {
+
+  const { updateImages, searchResults, openModal } = useGlobalContext()
+
+  // when location changes check for query param and execute the search
+  useEffect(() => {
+    const { search } = props.location
+    const { q } = queryString.parse(search)
+
+    if (q) {
+      searchPhotos(q).then(res => {
+        updateImages(res.data)
+      })
+    }
```

Comment:
I've found that `useEffect` is a super-powerful footgun, especially when it comes to data fetching. 

* Did you run into any issues with this data fetching pattern you used here on `Home.jsx`?
* Can you refactor this to use async/await syntax?

Reply:
I agree. You can get yourself into trouble with too many component re-renders and unexpected side-effects. In this case, it looks like the component does just render once when it mounts and then whenever search changes. Are there any other issues I might look out for here?  I did go ahead and use async/await though.